Plugins/NativeCamera/Source/NativeCameraImpl.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <bgfx/bgfx.h>
+#include <napi/napi.h>
+#include <arcana/threading/task.h>
+#include <NativeCamera.h>
+#include "CameraCapability.h"
+
+namespace Babylon::Plugins
+{
+    class CameraTrack final
+    {
+    public:
+        ~CameraTrack();
+
+        // Move semantics
+        CameraTrack(CameraTrack&&) noexcept;
+        CameraTrack& operator=(CameraTrack&&) noexcept;
+        
+        int32_t Width() const;
+        int32_t Height() const;
+        
+    private:
+        struct Impl;
+        CameraTrack(std::unique_ptr<Impl> impl);
+        std::unique_ptr<Impl> m_impl;
+        
+        friend class Camera::Impl;
+    };
+
+    class CameraDevice final
+    {
+    public:
+        ~CameraDevice();
+        
+        // Move semantics
+        CameraDevice(CameraDevice&&) noexcept;
+        CameraDevice& operator=(CameraDevice&&) noexcept;
+                
+        const std::vector<CameraTrack>& SupportedResolutions() const;
```

Comment:
@bghgary, this is the route I landed on for giving the consumer read access to the `std::vector<CameraTrack>` that the `CameraDevice` still owns the lifetime for.

I can try to work out the `std::vector<std::reference_wrapper<CameraTrack>>` but I think that route might require allocating a second vector.